FirebasePerformance/Sources/AppActivity/FPRSessionManager.m

```diff
@@ -57,51 +55,40 @@ - (FPRSessionManager *)initWithNotificationCenter:(NSNotificationCenter *)notifi
   self = [super init];
   if (self) {
     _sessionNotificationCenter = notificationCenter;
-    _trackingApplicationStateChanges = NO;
-    [self updateSessionId:nil];
+    _sessionDetails = [[FPRSessionDetails alloc] initWithSessionId:@"placeholder"
+                                                           options:FPRSessionOptionsNone];
   }
   return self;
 }
 
-- (void)startTrackingAppStateChanges {
-  if (!self.trackingApplicationStateChanges) {
-    // Starts tracking the application life cycle events during which the session Ids change.
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(updateSessionId:)
-                                                 name:UIApplicationWillEnterForegroundNotification
-                                               object:[UIApplication sharedApplication]];
-    self.trackingApplicationStateChanges = YES;
-  }
-}
-
-- (void)renewSessionIdIfRunningTooLong {
+- (void)stopGaugesIfRunningTooLong:(NSDate *)now {
   NSUInteger maxSessionLength = [[FPRConfigurations sharedInstance] maxSessionLengthInMinutes];
-  if (self.sessionDetails.sessionLengthInMinutes > maxSessionLength) {
-    [self updateSessionId:nil];
+  NSUInteger sessionLengthMinutes = [self.sessionDetails sessionLengthInMinutes:now];
+  if ([self.sessionDetails sessionLengthInMinutes:now] >= maxSessionLength) {
+    [[FPRGaugeManager sharedInstance] stopCollectingGauges:FPRGaugeCPU | FPRGaugeMemory];
   }
 }
 
 /**
- * Updates the sessionId on the arrival of a notification.
+ * Stops current session, and create a new session with new session id.
  *
- * @param notification Notification received.
+ * @param sessionIdString New session id.
  */
-- (void)updateSessionId:(NSNotification *)notification {
-  NSUUID *uuid = [NSUUID UUID];
-  NSString *sessionIdString = [uuid UUIDString];
-  sessionIdString = [sessionIdString stringByReplacingOccurrencesOfString:@"-" withString:@""];
-  sessionIdString = [sessionIdString lowercaseString];
+- (void)updateSessionId:(NSString *)sessionIdString {
+  [self updateSessionId:sessionIdString forceGauges:false];
+}
 
+- (void)updateSessionId:(NSString *)sessionIdString forceGauges:(BOOL)forceGauges {
```

Comment:
What is the purpose of the parameter `forceGauges`? Any situation where we force to collect gauges?

Reply:
It's for testing-only. I've removed it now and use `[OCMockObject partialMockForObject:...]` instead.